pulsar-client-cpp/include/pulsar/ConsumerConfiguration.h

```diff
@@ -33,6 +33,7 @@ class PulsarWrapper;
 
 /// Callback definition for non-data operation
 typedef boost::function<void(Result result)> ResultCallback;
+typedef boost::function<void(const Message& msg)> ReceiveCallback;
```

Comment:
then we should add request into pending-queue and it should complete the callback when the message will be available.